lib/admin/WP_Auth0_Admin_Features.php

```diff
@@ -291,24 +292,48 @@ public function render_override_wp_avatars( $args = array() ) {
 	}
 
 	public function basic_validation( $old_options, $input ) {
+		// SLO will be turned off in WP_Auth0_Admin_Features::sso_validation() if SSO is not on.
 		$input['singlelogout']        = ( isset( $input['singlelogout'] ) ? $input['singlelogout'] : 0 );
 		$input['override_wp_avatars'] = ( isset( $input['override_wp_avatars'] ) ? $input['override_wp_avatars'] : 0 );
 
 		return $input;
 	}
 
+	/**
+	 * Update the Auth0 Application if SSO is turned on and disable SLO if it is turned off.
+	 *
+	 * @param array $old_options - option values before saving.
+	 * @param array $input - new option values being saved.
+	 *
+	 * @return array
+	 */
 	public function sso_validation( $old_options, $input ) {
 		$input['sso'] = ( isset( $input['sso'] ) ? $input['sso'] : 0 );
-
-		if ( $old_options['sso'] != $input['sso'] && 1 == $input['sso'] ) {
-			if ( false === WP_Auth0_Api_Client::update_client( $input['domain'], $input['auth0_app_token'], $input['client_id'], $input['sso'] == 1 ) ) {
-
-				$error  = __( 'There was an error updating your Auth0 App to enable SSO. To do it manually, turn it ', 'wp-auth0' );
-				$error .= '<a href="https://auth0.com/docs/sso/single-sign-on#1">HERE</a>.';
-				$this->add_validation_error( $error );
-
+		$is_sso = ! empty( $input['sso'] );
```

Comment:
I don't want to change how the validation actually works (see the original) since it does work. I'm just forcing a boolean to check more clearly in this method.